app/models/sample.rb

```diff
@@ -179,11 +179,24 @@ def initiate_input_file_upload
   def initiate_s3_cp
     return unless status == STATUS_CREATED
     stderr_array = []
+    total_reads_json = "total_reads.json"
+    total_reads_json_path = nil
     input_files.each do |input_file|
       fastq = input_file.source
+      total_reads_json_path = File.join(File.dirname(fastq.to_s), total_reads_json)
       _stdout, stderr, status = Open3.capture3("aws", "s3", "cp", fastq.to_s, "#{sample_input_s3_path}/#{input_file.name}")
       stderr_array << stderr unless status.exitstatus.zero?
     end
+    if total_reads_json.present?
+      # For samples where we are only given fastas post host filtering, we need to input the total reads (before host filtering) from this file.
+      _stdout, _stderr, status = Open3.capture3("aws", "s3", "cp", total_reads_json_path, "#{sample_input_s3_path}/#{total_reads_json}")
+      # We don't have a good way to know if this file should be present or not, so we just try to upload it
+      # and if it fails, we try one more time;  if that fails too, it's okay, the file is optional.
```

Comment:
You can check whether file is present as follows?
```
_stdout, _stderr, status = Open3.capture3("aws", "s3", "ls", total_reads_json_path)
unless status.exitstatus.zero?
  ...
```

Reply:
Why would I want to do that?   If the file is present, I've just wasted an operation.  If the file is absent, I get an error the same way I would if I try to copy it.   Also, either approach can be a false negative for absence due to operational issues.   For this reason I am actually removing check_file_s3_presence from the python pipeline in https://github.com/chanzuckerberg/idseq-pipeline/pull/174

Reply:
Btw while working on the pipeline check_s3_file_presence this weekend I looked up the s3 api for that, something called head-object, with which you can distinguish more easily operational errors from file absence.  But it's too much work to be worth it at this time, I think.